in_toto/verifylib.py

```diff
@@ -635,6 +640,113 @@ def verify_all_item_rules(items, links, target_links=None):
         link.products, target_links)
 
 
+def verify_threshold_equality(layout, chain_link_dict):
+  """
+  <Purpose>
+    Iteratively verifies threshold conditions (if a step has been
+    performed and signed by enough number of functionaries) and link
+    artifacts (materials and products of a step is same for different
+    fucntionaries) of passed dictionary (chain_link_dict).
+
+  <Arguments>
+    layout:
+            The layout specified by the project owner against which the
+            threshold will be verified.
+
+    chain_link_dict:
+            A dictionary of key-link pair with step names as keys. For each
+            step name, there are one or more keyids and corresponding
+            link objects.
+
+  <Exceptions>
+    raises an Exception if threshold is not verified.
+    ThresholdVerificationError if the step is not performed by enough
+    functionaries or if the materials and products for a step are not same
+    for all functionaries.
+
+  <Side Effects>
+    None.
+
+  """
+
+  # We are only interested in links that are related to steps defined in the
+  # Layout, so iterate over layout.steps
+  for step in layout.steps:
+    # Skip steps that don't require multiple functionaries
+    if step.threshold <= 1:
+      continue
+
+    # Extract the key_link_dict for this step from the passed chain_link_dict
+    key_link_dict = chain_link_dict[step.name]
+
+    # take one exemplary link (e.g. the first in the step_link_dict)
+    compare_link = key_link_dict.values()[0]
+
+    # Check if we have at least <threshold> links for this step
+    if len(key_link_dict) < step.threshold:
+      raise ThresholdVerificationError("Step not performed" +
+        " by enough functionaries!")
+
+    # iterate over the remaining links to compare their properties
+    for keyid, link in six.iteritems(key_link_dict):
+      keys_dict = {keyid: layout.keys.get(keyid)}
+
+      # check if the authorized functionary has signed the link
+      try:
+        verify_link_signatures(link, keys_dict)
+      except KeyError as e:
```

Comment:
I think you don't want to catch a `KeyError` here but rather a `SignatureVerificationError`, right?